zip-0227.rst

```diff
@@ -118,20 +118,26 @@ This allows the issuer to use the same wallet it usually uses to transfer Assets
 Asset Identifier
 ----------------
 
-For every new Asset, there must be a new and unique Asset Identifier. We define this to be a globally unique pair :math:`(\mathsf{ik}, \mathsf{asset\_desc})`, where :math:`\mathsf{ik}` is the issuance key and :math:`\mathsf{asset\_desc}` is a byte string.
+Every asset must have a unique Asset Identifier. We define this to be a globally unique pair:
+:math:`(\mathsf{ik}, \mathsf{asset\_desc})`, where: :math:`\mathsf{ik}` is the issuance public key and :math:`\mathsf{asset\_desc}` is a UTF-8 byte string.
 
-A given Asset Identifier is used across all Zcash protocols that support ZSAs -- that is, the Orchard-based ZSA protocol and potentially future Zcash shielded protocols. For this Asset Identifier, we derive an Asset Digest, :math:`\mathsf{AssetDigest}`, which is simply is a :math:`\mathsf{BLAKE2b-512}` hash of the Asset Identifier.
-From the Asset Digest, we derive a specific Asset Base within each such shielded protocol (for example :math:`\mathsf{AssetBase}^{\mathsf{Orchard}}_{\mathsf{AssetId}}` for the Orchard-based ZSA protocol), using the applicable hash-to-curve algorithm. This Asset Base is included in shielded notes.
+A given Asset Identifier is used across all Zcash protocols that support ZSAs.
+For this Asset Identifier, we derive an Asset Digest, :math:`\mathsf{AssetDigest}`, which is :math:`\mathsf{BLAKE2b-256}` hash of the Asset Identifier.
+Then, a specific Asset Base is derived from the Asset Digest within each shielded protocol.
+
+For Orchard: 
+:math:`\mathsf{AssetBase}^{\mathsf{Orchard}}_{\mathsf{AssetId}}` , using the applicable hash-to-curve algorithm. This Asset Base is included in shielded notes.
 
 Let 
 
 - :math:`\mathsf{asset\_desc}` be the asset description, a UTF-8 encoded string of up to 512 bytes, which includes any information pertaining to the issuance.
 - :math:`\mathsf{ik}` be the issuance validating key of the issuer, a public key used to verify the signature on the issuance transaction's SIGHASH.
 
-Define :math:`\mathsf{AssetDigest_{\mathsf{AssetId}}} := \mathsf{BLAKE2b\text{-}512}(\mathsf{EncodeAssetId}(\mathsf{AssetId}))`,
+Define :math:`\mathsf{AssetDigest_{\mathsf{AssetId}}} := \mathsf{BLAKE2b\text{-}256}(\mathsf{EncodeAssetId}(\mathsf{AssetId}))`,
+
```

Comment:
Also, a personalaization for the hash is required:
```rust
    /// Personalization for the ZSA asset digest generator
    pub const ZSA_ASSET_DIGEST_PERSONALIZATION: &[u8; 22] = b"z.cash:ZSA-AssetDigest";

```

Reply:
![image](https://user-images.githubusercontent.com/3682187/218301724-484421a2-0185-4146-9741-d60fd2158087.png)

Reply:
![image](https://user-images.githubusercontent.com/3682187/218302250-0abd97a4-8c49-45ed-b298-97f193265b83.png)

Reply:
Will add the personalization, yes. Will let you know if I think of any other options for the string. Also, see [this comment](https://github.com/zcash/zips/pull/649#discussion_r1107744727) about using BLAKE2b-512 instead of BLAKE2b-256.